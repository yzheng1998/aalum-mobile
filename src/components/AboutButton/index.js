import React from 'react'
import LinearGradient from 'react-native-linear-gradient'
import { Wrapper, ButtonText, gradientStyle } from './styles'
import theme from '../../../theme'

const { gradientLeft, gradientRight } = theme.colors

const AccountButton = ({ text, color, disabled, gradient, borderColor }) => (
  <Wrapper borderColor={borderColor}>
    {gradient ? (
      <LinearGradient
        start={{ x: 0, y: 0 }}
        end={{ x: 1, y: 0 }}
        colors={[gradientLeft, gradientRight]}
        style={{
          ...gradientStyle,
          opacity: disabled ? 0.6 : 1,
          flexDirection: 'row'
        }}
      >
        <ButtonText color={color}>{text}</ButtonText>
      </LinearGradient>
    ) : (
      <ButtonText color={color}>{text}</ButtonText>
    )}
  </Wrapper>
)

export default AccountButton
