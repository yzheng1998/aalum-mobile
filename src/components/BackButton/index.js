import React, { Component } from 'react'
import Icon from '../Icon'
import { IconWrapper } from './styles'

export default class BackButton extends Component {
  render() {
    return (
      <IconWrapper onPress={this.props.onPress}>
        <Icon name="arrow-back" color="black" size={30} />
      </IconWrapper>
    )
  }
}
