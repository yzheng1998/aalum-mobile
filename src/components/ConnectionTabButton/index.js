import React from 'react'
import {
  InactiveButton,
  ActiveButton,
  Wrapper,
  ButtonText,
  Icon
} from './styles'

const ConnectionTabButton = ({ icon, text, active }) => (
  <Wrapper>
    {active ? (
      <ActiveButton>
        <Icon source={icon} />
      </ActiveButton>
    ) : (
      <InactiveButton>
        <Icon source={icon} />
      </InactiveButton>
    )}
    <ButtonText active={active}>{text}</ButtonText>
  </Wrapper>
)

export default ConnectionTabButton
