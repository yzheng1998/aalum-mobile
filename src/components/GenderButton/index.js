import React, { Component } from 'react'
import { Button, ButtonText } from './styles'

export default class GenderButton extends Component {
  render() {
    const { title, selectGender, selected, disabled } = this.props
    return (
      <Button disabled={disabled} onPress={selectGender} selected={selected}>
        <ButtonText selected={selected}>{title}</ButtonText>
      </Button>
    )
  }
}
