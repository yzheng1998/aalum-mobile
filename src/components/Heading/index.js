import styled from 'styled-components'
import { themeGet } from 'styled-system'

export const Heading = styled.Text`
  font-size: 16px;
  font-family: ${themeGet('fonts.heading')};
  color: ${themeGet('colors.charcoal')};
`
