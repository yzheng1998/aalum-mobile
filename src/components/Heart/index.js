import React, { Component } from 'react'
import { Icn } from './styles'

class Heart extends Component {
  render() {
    const { active } = this.props
    return <Icn active={active} name={active ? 'heart' : 'heart-empty'} />
  }
}

export default Heart
