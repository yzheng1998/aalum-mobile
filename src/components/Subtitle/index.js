import styled from 'styled-components'
import theme from '../../../theme'

export const Subtitle = styled.Text`
  align-self: center;
  text-align: center;
  font-size: 18px;
  margin-top: 16px;
  margin-left: 36px;
  margin-right: 36px;
  margin-bottom: 30px;
  font-family: ${theme.fonts.bodyText};
  color: ${theme.colors.grey};
`
