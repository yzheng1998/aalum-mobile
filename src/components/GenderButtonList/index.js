import React, { Component } from 'react'
import { ButtonContainer } from './styles'
import GenderButton from '../GenderButton'

export default class GenderButtonList extends Component {
  render() {
    const { genderSelection, selectGender, allSelected, disabled } = this.props
    return (
      <ButtonContainer>
        {genderSelection.map((gender, index) => (
          <GenderButton
            key={gender.title}
            title={gender.title}
            selectGender={() => selectGender(index)}
            selected={allSelected || gender.selected}
            disabled={disabled && !gender.selected}
          />
        ))}
      </ButtonContainer>
    )
  }
}
