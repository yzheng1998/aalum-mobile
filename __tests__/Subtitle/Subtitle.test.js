import React from 'react'
import { Subtitle } from '../../src/components/Subtitle'
import mountWithTheme from '../helpers/theme'

describe('Subtitle', () => {
  test('renders correctly', () => {
    const wrapper = mountWithTheme(<Subtitle>Title!</Subtitle>)
    expect(wrapper).toMatchSnapshot()
  })
})
