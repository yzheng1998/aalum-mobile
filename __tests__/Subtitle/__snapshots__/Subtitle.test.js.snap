// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle renders correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <Styled(Text)>
    <StyledNativeComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(Text)",
          "inlineStyle": InlineStyle {
            "rules": Array [
              "
  align-self: center;
  text-align: center;
  font-size: 18px;
  margin-top: 16px;
  margin-left: 36px;
  margin-right: 36px;
  margin-bottom: 30px;
  font-family: ",
              "Avenir-Medium",
              ";
  color: ",
              "#A8A8A8",
              ";
",
            ],
          },
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#A8A8A8",
              "fontFamily": "Avenir-Medium",
              "fontSize": 18,
              "marginBottom": 30,
              "marginLeft": 36,
              "marginRight": 36,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#A8A8A8",
                "fontFamily": "Avenir-Medium",
                "fontSize": 18,
                "marginBottom": 30,
                "marginLeft": 36,
                "marginRight": 36,
                "marginTop": 16,
                "textAlign": "center",
              },
            ]
          }
        >
          Title!
        </Text>
      </Text>
    </StyledNativeComponent>
  </Styled(Text)>
</ThemeProvider>
`;
